### ISSUE TITLE
Only timestamp and <nil> output sometimes on export

### ISSUE THREAD

**Issue by DI-DaveGoodine** [jonasvinther/medusa#43]
os: mac os 11.2.3
binary downloaded from here: https://github.com/jonasvinther/medusa/releases/download/v0.2.0/medusa_0.2.0_darwin_amd64.tar.gz
vault: 1.5.3

Steps:
Set VAULT_ADDR, and run export:
``` ./medusa export di-secrets/ -t s.tokenstuff --format="yaml" --insecure```
output:
```2021/04/05 21:04:47 <nil> <nil>```

Expected:
All the secret data in the engine at di-secrets (v2) should be exported to stdout.

Notes:
* di-secrets is a v2 kv store
* some exports work, for example: ```./medusa export di-secrets/application -t s.foo```

**Comment by jonasvinther** [comment]
Hi and thank you for reaching out!

I haven't seen this issue before. Are there any patterns in the secrets that is unusual? For instance special characters or  multi line value or something like that.
It is also a bit strange that you are saying that some of your exports are working fine. Is it the same export that is failing every time or is it random?
If possible I would like to see an example of the data that is causing the issue.

**Comment by hoeghh** [comment]
Hi @DI-DaveGoodine 

Does the policy assigned to the token used have access to all data under the `di-secrets` engine? Or are there some elements there that it doesnt have access to?

**Comment by DI-DaveGoodine** [comment]
> Hi and thank you for reaching out!
> 
> I haven't seen this issue before. Are there any patterns in the secrets that is unusual? For instance special characters or multi line value or something like that.
> It is also a bit strange that you are saying that some of your exports are working fine. Is it the same export that is failing every time or is it random?
> If possible I would like to see an example of the data that is causing the issue.

@jonasvinther It seems to be the same export - my use case is exporting all the secrets in an engine for DR purposes.

Unfortunately, I don't have a good test data set - I am trying to export our Prod secrets.

**Comment by DI-DaveGoodine** [comment]
> Hi @DI-DaveGoodine
> 
> Does the policy assigned to the token used have access to all data under the `di-secrets` engine? Or are there some elements there that it doesnt have access to?

@hoeghh I used a token that has access to everything.

**Comment by hoeghh** [comment]
Hi @DI-DaveGoodine 

Does the token have the root policy? Could you check with a root token? Just to make sure it's not a permission thing. 
My guess is that if you dont have list permission on the secret engines root, it might be the problem.

**Comment by jonasvinther** [comment]
I have tested this and the token needs the following policy:
```
path "secret-engine/*" {
  capabilities = ["list", "read"]
}
```

**Comment by jonasvinther** [comment]
I have tested this and the token needs the following policy:
```
path "secret-engine/*" {
  capabilities = ["list", "read"]
}
```
@DI-DaveGoodine can you verify if this is the case?

**Comment by DI-DaveGoodine** [comment]
> I have tested this and the token needs the following policy:
> 
> ```
> path "secret-engine/*" {
>   capabilities = ["list", "read"]
> }
> ```
> 
> @DI-DaveGoodine can you verify if this is the case?

@jonasvinther , @hoeghh  - I use `vault login` to get a token, and that token has these policies:
policies               ["default" "superuser"]

superuser is:
```
# grant full access to all Vault paths
path "*" {
    capabilities = ["create", "read", "update", "delete", "list", "sudo"]
}
```

**Comment by oshamin** [comment]
I faced the same error.
```
# ./medusa export secret/ --format="json" --insecure -m kv1
2021/04/14 21:13:26 <nil> <nil>
```
I use the initial root token in environment variable.

**Comment by jonasvinther** [comment]
@oshamin  are you also running Mac os?

**Comment by oshamin** [comment]
@jonasvinther no, It runs on linux (ubuntu 16.0).

**Comment by jonasvinther** [comment]
Hi @DI-DaveGoodine  and @oshamin 
I have now found the location in the code where this error is happening. It's happening while reading a secret. So I suspect that I somehow have messed up the vault path for some secrets and therefore Medusa is unable to read the secret. I have added some debugging code that should tell more about what is actually happening. 

Can you please download and run Medusa version 0.2.1 and then post the error message in this chat?
You should then get a error message that looks something like this. Remember to anonymize your data before posting!
```
2021/04/15 22:01:21  Error while reading secret
Path:   secret/data/A/B/C/D
Data:   map[string]interface {}{"some-key":"some value"}
```

**Comment by DI-DaveGoodine** [comment]
@jonasvinther here is the result:
```
davegoodine@HOU-L9452: medusa_0.2.1_darwin_amd64$ ./medusa export di-secrets/ -t s.foo --format="yaml" --insecure
2021/04/15 15:58:15 Error while reading secret
Path:	di-secrets/data//cds-gcb/application/gdm/gcb-amq-pass-rio-tinto-dev-ew1
Data:	<nil>
```
When I view the secret at that path in the UI, I see:

>  Version 2 of this secret has been deleted

> A version that has been deleted but can be undeleted using the Version 2 menu above. You can also see other versions of this secret in the History menu.

**Comment by oshamin** [comment]
Here is 5 cent from me.
```
# ./medusa export secret --format="json" --insecure -m kv1
2021/04/16 11:17:43 Error while reading secret
Path:   secret//********/********/environments/dev/deployment/image-pusher-dev
Data:   <nil>
```

**Comment by jonasvinther** [comment]
Hi @DI-DaveGoodine and @oshamin 
I have released a v0.2.2. I hope that this will fix your bugs. 

For the kv2 issue that you have @DI-DaveGoodine it's simply because, as you also wrote, that the secret has no active version because the version of the secret has been deleted. And since kv2 keep's the history until you delete the secret and not just a version, you would get the error you did. I now handle this by simply marking the secret as empty in the export because this is the most correct way to handle a secret with no active version I think. 

For the kv1 issue this was simply because I was handling the data incorrectly. I actually got a bit of help to solve this one in a pull request #45.

Please let me know if these changes solves the issues. And thank you for the patience and feedback.

**Comment by DI-DaveGoodine** [comment]
@jonasvinther  latest version works great, thank you!
